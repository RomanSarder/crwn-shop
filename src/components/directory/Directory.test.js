import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import { createMemoryHistory } from 'history'
import DirectoryContainer, { Directory } from './Directory'

jest.mock('../menu-item/MenuItem', () => () => (<div>Hello World</div>))

const testUrl = 'test-url'
var testSections = [
    {
        title: 'hats', 
        imageUrl: 'test-url-1', 
        id: 1, 
        size: 'large' 
    },
    {
        title: 'mens',
        imageUrl: 'test-url-2',
        id: 2,
    }
]

function renderDirectory () {
    var renderResult = render(
        <Directory sections={testSections} url={testUrl} />
    )

    return renderResult
}

function renderDirectoryContainer (store) {
    var history = createMemoryHistory()
    history.push(testUrl)

    var renderResult = render(
        <Router history={history}>
            <Provider store={store}>
                <DirectoryContainer />
            </Provider>
        </Router>
    )

    return renderResult
}

it('<Directory /> should properly render with mocked menu items', () => {
    var { container, getAllByText } = renderDirectory()

    expect(container.firstChild).toMatchSnapshot()
    expect(getAllByText('Hello World').length).toEqual(2)
})

it('<Directory /> should properly render with redux state', () => {
    var store = configureStore()({
        directory: {
            items: testSections
        }
    })

    var { container, getAllByText } = renderDirectoryContainer(store)
    expect(container.firstChild).toMatchSnapshot()
    expect(getAllByText('Hello World').length).toEqual(2)
})