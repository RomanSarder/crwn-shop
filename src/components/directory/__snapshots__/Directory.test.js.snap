// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Directory /> should properly render with mocked menu items 1`] = `
<div
  class="sc-bdvvaa wpwIC"
>
  <div>
    Hello World
  </div>
  <div>
    Hello World
  </div>
</div>
`;

exports[`<Directory /> should properly render with redux state 1`] = `
<div
  class="sc-bdvvaa wpwIC"
>
  <div>
    Hello World
  </div>
  <div>
    Hello World
  </div>
</div>
`;
